File: 0290-word-pattern/0290-word-pattern.java
class Solution {
    public boolean wordPattern(String pattern, String s) {
        String[] sArr = s.split(" "); // -> apple apple banana banana to array

        HashMap<Character, String> map = new HashMap<>();

        if(sArr.length!= pattern.length()) {
            return false;
        }

        for(int i=0; i<sArr.length; i++){
            char c = pattern.charAt(i);

            if(map.containsKey(c)){
                if(!map.get(c).equals(sArr[i])) {
                    return false;
                }
            } else {
                if(map.containsValue(sArr[i])) {
                    return false;
                }

                map.put(c, sArr[i]);
            }
        }

        return true;
    }
}

File: 0290-word-pattern/0290-word-pattern.kt
class Solution {
fun wordPattern(pattern: String, s: String): Boolean {
    val list = s.split(" ")
    if (list.size != pattern.length) return false

    val map = mutableMapOf<Char, String>()
    pattern.forEachIndexed { i, ch ->
        when (map.containsKey(ch)) {
            true -> if (map[ch] != list[i]) return false
            false -> {
                if (map.containsValue(list[i])) return false
                map[ch] = list[i]
            }
        }
    }

    return true
}
}